### ISSUE TITLE
Flat values are converted in parse_string API

### ISSUE THREAD

**Issue by komalb007** [Z3Prover/z3#3037]
Hi Experts,

I am seeing a behavior in parse_string API.
My use case is ,
I am giving Z3 expression string as 
**(assert( => (= A "A1") (and (> D 1.8) (< D 3.9))))**
which is checking for range for value of "D" when A = A1

When I am passing this to _ctx.parse_string_ API to give me Z3 expression, 
It is returning me the expression as 
**(assert (=> (= A "A1") (and (> D (/ 9.0 5.0)) (< D (/ 39.0 10.0)))) )**

Basically it is modifying the float values as per some calculations, I am not able to understand the pattern here.
Why is it happening?
Is there any way to stop it?
I am comparing two strings and they are different though they are not.

Thanks,

**Comment by NikolajBjorner** [comment]
The internal representation of decimal numbers is as rational numbers. 
This is what gets printed.

**Comment by komalb007** [comment]
Okay..
Is there any way I can get string as is after parse_string ?
I am getting this string as below,

    std::stringstream currentassertstr;
    currentassertstr << "(assert ";
    currentassertstr << z3expr;
    currentassertstr << " )";

As I am doing comparison between these two and they are not equal.

**Comment by NikolajBjorner** [comment]
I don't think so
